doc/reference/configuration/configuration_reference.rst

```diff
@@ -221,6 +488,718 @@ This section describes options related to :ref:`storing configuration in etcd <c
+
+    UUIDs should follow these rules:
+
+    *   The values must be true unique identifiers, not shared by other instances
+        or replica sets within the common infrastructure.
+
+    *   The values must be used consistently, not changed after the initial setup.
+        The initial values are stored in :ref:`snapshot files <index-box_persistence>`
+        and are checked whenever the system is restarted.
+
+    *   The values must comply with `RFC 4122 <https://tools.ietf.org/html/rfc4122>`_.
+        The `nil UUID <https://tools.ietf.org/html/rfc4122#section-4.1.7>`_ is not allowed.
+
+    See also: :ref:`database.replicaset_uuid <configuration_reference_database_replicaset_uuid>`
+
+    | Type: string
+    | Default: box.NULL
+    | Environment variable: TT_DATABASE_INSTANCE_UUID
+
+
+.. _configuration_reference_database_mode:
+
+.. confval:: database.mode
+
+    An instance's operating mode.
+    The following modes are available:
+
+    -   ``rw``: an instance is in read-write mode.
+    -   ``ro``: an instance is in read-only mode.
+
+    If this option is not specified explicitly, its default value depends on the number of instances in a replica set. For a single instance, the ``rw`` mode is used, while for multiple instances, the ``ro`` mode is used.
+
+    **Example**
+
+    You can set the ``database.mode`` option to ``rw`` on all instances in a replica set to make a :ref:`master-master <replication-roles>` configuration.
+    In this case, :ref:`replication.failover <configuration_reference_replication_failover>` should be set to ``off``.
+
+    ..  literalinclude:: /code_snippets/snippets/replication/instances.enabled/master_master/config.yaml
+        :language: yaml
+        :dedent:
+
+    | Type: string
+    | Default: box.NULL
+    | Environment variable: TT_DATABASE_MODE
+
+
+.. _configuration_reference_database_replicaset_uuid:
+
+.. confval:: database.replicaset_uuid
+
+    A :ref:`replica set UUID <replication_uuid>`.
+
+    By default, replica set UUIDs are generated automatically.
+    ``database.replicaset_uuid`` can be used to specify a replica set identifier manually.
+
+    See also: :ref:`database.instance_uuid <configuration_reference_database_instance_uuid>`
+
+    | Type: string
+    | Default: box.NULL
+    | Environment variable: TT_DATABASE_REPLICASET_UUID
+
+
+.. _configuration_reference_database_txn_isolation:
+
+.. confval:: database.txn_isolation
+
+    A transaction :ref:`isolation level <transaction_model_levels>`.
+
+    | Type: string
+    | Default: ``best-effort``
+    | Possible values: ``best-effort``, ``read-committed``, ``read-confirmed``
+    | Environment variable: TT_DATABASE_TXN_ISOLATION
+
+
+.. _configuration_reference_database_txn_timeout:
+
+.. confval:: database.txn_timeout
+
+    A timeout (in seconds) after which the transaction is rolled back.
+
+    See also: :ref:`box.begin() <box-begin>`
+
+    | Type: number
+    | Default: 3153600000
```

Comment:
I think that such large numbers in seconds requires some short explanations, like so:

> Default: 3153600000 (~100 years)